### PR TITLE
feat: add HubInitiative class

### PR DESCRIPTION
**Pull request by dbouwman**
1. Description:

- add HubInitiative Class
- hoist repetitive PropertyMap code to central function
- hoist common functionality to  abstract `HubItemEntity` class, which `HubProject` and `HubInitiative` extend. This is setting up for additional "common" features like resource and thumbnail management
- implement `canEdit` and `canDelete` on `HubItemEntity` and add tests

1. Instructions for testing:
- run tests

1. Closes Issues: [4658](https://devtopia.esri.com/dc/hub/issues/4658)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

### CODE REVIEW

File: packages/common/src/core/types/IHubContent.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export enum PublisherSource {
  * Data model for content
  */
 export interface IHubContent
-  extends IHubItemEntity,
+  extends Omit<IHubItemEntity, "schemaVersion">,
```
Comment (dbouwman) [inline packages/common/src/core/types/IHubContent.ts:22]: This allows `schemaVersion` to be added to `IHubItemEntity` but to not be part of `IHubContent`

File: packages/common/src/projects/HubProject.ts [hunk 2]
```diff
@@ -20,27 +18,19 @@ import {
...
-  unshareItemWithGroup,
-} from "@esri/arcgis-rest-portal";
-import { sharedWith } from "../core/_internal/sharedWith";
+import { HubItemEntity } from "../core/HubItemEntity";
 
 /**
  * Hub Project Class
  */
 export class HubProject
+  extends HubItemEntity<IHubProject>
```
Comment (dbouwman) [inline packages/common/src/projects/HubProject.ts:27]: Able to drop a lot of code from this file by extending from the base

**Review by benstoltz — APPROVED** [review]
🙇 Great work.

**Review by mjuniper — APPROVED** [review]
Didn't dig deeply but at a high level this LGTM.

File: packages/common/src/core/types/IHubContent.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ export enum PublisherSource {
  * Data model for content
  */
 export interface IHubContent
-  extends IHubItemEntity,
+  extends Omit<IHubItemEntity, "schemaVersion" | "extent">,
```
Comment (dbouwman) [inline packages/common/src/core/types/IHubContent.ts:22]: @tomwayson Note the `Omit<... "extent">` which was needed b/c I added `extent: IExtent` to the `IHubItemEntity`

Comment (tomwayson) [reply packages/common/src/core/types/IHubContent.ts:22]: why not have `IHubItemEntity.extent: BBox` so that it aligns w/  `IItem`?

Comment (dbouwman) [reply packages/common/src/core/types/IHubContent.ts]: Changed to `number[][]` which is what `IItem.extent` is defined as

**Review by tomwayson — APPROVED** [review]

**Pull request merged**